src/pages/fieldManagement/settings/defaultFieldComponents/RelationshipEditor.jsx

```diff
@@ -0,0 +1,235 @@
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+import { get } from 'lodash-es';
+
+import IconButton from '@material-ui/core/IconButton';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import NewChildIcon from '@material-ui/icons/AddCircle';
+import TextInput from '../../../../components/inputs/TextInput';
+import DeleteButton from '../../../../components/DeleteButton';
+import Button from '../../../../components/Button';
+import Text from '../../../../components/Text';
+
+function deleteCategory(allRelationships, category) {
+  delete allRelationships[category];
+  return allRelationships;
+}
+
+function deleteRelationship(
+  allRelationships,
+  currentCategory,
+  relationship,
+) {
+  const targetRelationshipCategory = get(
+    allRelationships,
+    currentCategory,
+  );
+  const relationshipsWithoutTarget = targetRelationshipCategory.filter(
+    entry => entry !== relationship,
+  );
+  allRelationships[currentCategory] = relationshipsWithoutTarget;
+  return allRelationships;
+}
+
+function updateRelationshipInCategory(
+  allRelationships,
+  category,
+  oldRelationshipName,
+  newRelationshipName,
+) {
+  const currentRelationships = get(allRelationships, category);
+
+  const oldRelationshipIndex = currentRelationships?.indexOf(
+    oldRelationshipName,
+  );
+  if (oldRelationshipIndex > -1)
+    currentRelationships[oldRelationshipIndex] = newRelationshipName;
+  allRelationships[category] = currentRelationships;
+  return allRelationships;
+}
+
+function createCategory(allRelationships) {
+  console.log(allRelationships);
+  allRelationships[''] = [];
+  return allRelationships;
+}
+
+function updateCategoryName(
+  allRelationships,
+  currentCategoryName,
+  newCategoryName,
+) {
+  const relationshipNames = get(
+    allRelationships,
+    currentCategoryName,
+  );
+  delete allRelationships[currentCategoryName];
+  allRelationships[newCategoryName] = relationshipNames;
+  return allRelationships;
+}
+
+function addRelationshipToCategory(
+  allRelationships,
+  currentCategory,
+  newRelationship,
+) {
+  const currentRelationships = get(allRelationships, currentCategory);
+  if (!currentRelationships.includes(newRelationship))
+    currentRelationships.push(newRelationship);
```

Comment:
same thing with arrays, instead of pushing to arrays it's usually better to create a new array with the item added

Reply:
Ok. What do you think of what I did here with the new commit?